### PR TITLE
Cleanup Phase I

### PR DESCRIPTION
**Pull request by the-maldridge**
This branch includes the first stage of cleanup efforts.  The short highlights include:
  * cleanup on variable naming
  * migrate common functionality to lib.sh
    * chroot access is now "automagic" across architectures
    * routes for future support to build x86 from foreign architectures are present
    * lib.sh encapsulates several common utility tasks such as retrieving the arch for a particular platform
  * much work to get passing shellcheck on mkrootfs.sh.in, mkplatformfs.sh.in, mkimage.sh.in; ultimately the goal is to turn on a travis task to run shellcheck against PRs

Several architectural changes have also been included in this branch:
  * build-arm-images.sh.in and build-rootfs.sh.in are gone
  * rootfs, platformfs, and images are built by makefile, this has been tested up to -j4
  * a base set of rootfs tarballs are used to derive all images (this will be expanded to mklive and mknet in the next phase of cleanup)

Future improvements that are scheduled include building the live images with the makefile for better parallelism, using the makefile to automate signing, and paving the way for a totally automated release pipeline.  Much further down the road I intend to take on the installer and try to clean it up a lot, its very ugly as is and requires a lot of binaries that can likely be done without (important for PXE environments).

### CODE REVIEW

File: lib.sh.in [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+    register_binfmt
+
+    # Before we step into the chroot we need to make sure the
+    # pseudo-filesystems are ready to go.  Not all commands will need
+    # this, but its still a good idea to call it here anyway.
+    mount_pseudofs
+
+    # With assurance that things will run now we can jump into the
+    # chroot and run stuff!
+    chroot "$1" sh -c "$2"
```
Comment (Gottox) [inline lib.sh.in]: I had the idea to use xbps-uchroot instead of chroot, this unshares the mountspace and bindmounts /dev /prov /sys. Unfortunately iirc there were some issues with it that needed to be fixed in uchroot.

File: mkimage.sh.in [hunk 2]
```diff
@@ -86,9 +81,10 @@ _EOF
...
 }
 
-#
-# main()
-#
+# ########################################
+#      SCRIPT EXECUTION STARTS HERE
+# ########################################
+
 while getopts "b:B:o:r:s:hV" opt; do
```
Comment (Gottox) [inline mkimage.sh.in]: Maybe we can use the getopt/getopts combination we use in xbps-src here too.

File: lib.sh.in (hunk 1 above)
Comment (the-maldridge) [inline lib.sh.in]: Does xbps-uchroot play nice with qemu-user-static?

Comment (Gottox) [inline lib.sh.in]: xbps-uchroot is actually chroot + unshare. I don't know if there are implications to qemu

File: mkimage.sh.in (hunk 2 above)
Comment (the-maldridge) [inline mkimage.sh.in]: I looked into this and it seems that the important options are being mirrored across.  I don't think copying over more options that are already copied over buys us anything.  I certainly don't think the dance that xbps-src does with preprocessing is worth it here.

File: Makefile [hunk 3]
```diff
@@ -23,6 +23,7 @@ ALL_CLOUD_IMAGES=$(foreach cloud,$(CLOUD_IMGS),void-$(cloud)-$(DATE).tar.gz)
 SUDO := sudo
 
 XBPS_REPOSITORY := -r https://lug.utdallas.edu/mirror/void/current -r https://lug.utdallas.edu/mirror/void/current/musl -r https://lug.utdallas.edu/mirror/void/current/aarch64
+COMPRESSOR_THREADS=2
```
Comment (Gottox) [inline Makefile]: Can we dynamicly use the number of CPUs in the system? Does this make sense?

And I'm curious: Does running multithreaded xz result always result in an identical file with the same checksum?

Comment (the-maldridge) [inline Makefile]: setting this to 0 will dynamically use the maximum number of threads.  The problem with this is that it also takes a lot of memory to do this.

Comment (Gottox) [inline Makefile]: Ok.

File: lib.sh.in (hunk 1 above)
Comment (Gottox) [inline lib.sh.in]: I think we should keep that change in our mind, But it shouldn't be part of this PR.

**Comment by Gottox** [comment]
Ok to merge for me.
